utils/kubernetes/service.go

```diff
@@ -73,5 +75,20 @@ func GetEndpoint(ctx context.Context, opts *ServiceOptions, obj *corev1.Service)
 		}
 	}
 
+	if !utils.TcpCheck(endpoint.External) {
+		if len(strings.SplitAfter(opts.APIServerURL, "://")) > 1 {
```

Comment:
I find it a bit difficult to understand the logic here, as I don't know enough about why `APIServerURL` is used in some cases (something specific to `minikube`). It would really be helpful to have an example of a `APIServerURL` here that would result in a valid endpoint.

Reply:
So in case of Minikube hyper kit clusters, the LoadBalancer IP turns out to be the same as that of the ClusterIP. So in this scenario it means that the LoadBalancer IP is nothing but the APIServer's IP.So here we strip off the IP and check if that's reachable, does this make sense? @mgfeller